functions/src/api/v1/users/getUser.ts

```diff
@@ -0,0 +1,57 @@
+import { Request, Response } from 'express'
+import { ForbiddenError } from '../../errors/HttpError'
+import { getUserDb } from '../../../db/getUserDb'
+
+/**
+ * @swagger
+ * /v1/users/{userId}:
+ *   get:
+ *     description: Get user details, subscription and quota status. This will not send any payment details or API keys.
+ *     tags: ['users']
+ *     produces:
+ *       - application/json
+ *     responses:
+ *       200:
+ *         description: User details
+ *         content:
+ *           application/json:
+ *             example: {
+ *               id: 'vXIUuaGkH4kukbwRH5cU',
+ *               updatedAt: 1605969562,
+ *               subscription: {
+ *                   isActive: true,
+ *                   isQuotaReached: false
+ *               },
+ *               sentEmails: 1,
+ *               sentSMSs: 2
+ *             }
+ *             schema:
+ *               type: object
+ *               $ref: '#/components/schemas/User'
+ *       401:
+ *         description: missing authorization bearer token
+ *       403:
+ *         description: authorization header present but not valid
+ *       412:
+ *         description: authorization header present but not formatted correctly
+ */
+export const getUser = async (req: Request, res: Response) => {
+    const currentUid = res.locals.uid
+    const requestedUserId = req.params.userId
+
+    if (requestedUserId !== currentUid) {
+        throw new ForbiddenError('Not authorized to get this resource')
+    }
+
+    const user = await getUserDb(currentUid)
+
+    res.send({
```

Comment:
wouldn't `res.json()` be more appropriate when sending the response to the client, since it also sets the response type ?

Reply:
the response type is send automatically using res.send with an object

Reply:
Are we sure about this? Pls give a look [here](https://www.tutorialspoint.com/difference-between-res-send-and-res-json-in-express-js)

Reply:
I am, 
![Capture d’écran 2020-12-14 à 13 52 16](https://user-images.githubusercontent.com/662377/102083337-aa749580-3e13-11eb-9868-387838b9e370.png)